components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/processor/JWTInboundRequestProcessor.java

```diff
@@ -0,0 +1,669 @@
+                    }
+                    String remoteClaimUri = entry.getKey().getRemoteClaim().getClaimUri();
+                    String localClaimUri = entry.getKey().getLocalClaim().getClaimUri();
+                    String claimValue = entry.getValue();
+                    claimsSet.claim(remoteClaimUri, claimValue);
+                    if (log.isDebugEnabled()) {
+                        log.debug("Adding user claim for local URI: " + neutralize(localClaimUri) +
+                                " with remote URI: " + neutralize(remoteClaimUri) + " and the value: " +
+                                neutralize(claimValue));
+                    }
+                }
+            }
+
+            SignedJWT signedJWT = new SignedJWT(new JWSHeader(this.jwsAlgorithm), claimsSet.build());
+            JWSSigner signer = new MACSigner(apiKey);
+            signedJWT.sign(signer);
+
+            String jwtToken = signedJWT.serialize();
+            if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable(JWTInboundConstants.IdentityTokens.JWT_TOKEN)) {
+                log.debug("JWT token generated: " + neutralize(jwtToken));
+            }
+            return jwtToken;
+        } catch (NumberFormatException e) {
+            throw new JWTIdentityException("Error while reading the token expiry time configuration", e);
+        } catch (KeyLengthException e) {
+            throw new JWTIdentityException("Error while generating JWT token signer", e);
+        } catch (JOSEException e) {
+            throw new JWTIdentityException("Error while signing JWT token", e);
+        }
+    }
+
+    /**
+     * This method is used to validate a string against a regex
+     *
+     * @param regex The regex
+     * @param input The string to be validated against the regex
+     * @return True if the validation is success.
+     */
+    public boolean validateRegexInput(String regex, String input) {
+
+        if (StringUtils.isNotBlank(regex) && StringUtils.isNotBlank((input))) {
+            if (log.isDebugEnabled()) {
+                log.debug("Validating regex for the input: " + neutralize(input) + " against the regex: " +
+                        neutralize(regex));
+            }
+            Pattern pattern = Pattern.compile(regex);
+            Matcher matcher = pattern.matcher(input);
+            if (matcher.matches()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Regex validation success.");
+                }
+                return true;
+            }
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Regex validation failed for the input: " + neutralize(input));
+        }
+        return false;
+    }
+
+    /**
+     * The method used to prevent CRLF_INJECTION_LOGS. It neutralizes the given input.
+     *
+     * @param input The input string to be neutralized.
+     * @return neutralized output.
+     */
+    private String neutralize(String input) {
```

Comment:
This method is defined in multiple classes. Define in a common class and reuse

Reply:
Fixed